clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -258,7 +258,10 @@ public void poll(long now) {
         }
 
         pollHeartbeat(now);
-        maybeAutoCommitOffsetsAsync(now);
+        if (subscriptions.hasNoSubscription())
+            subscriptions.clearAssignment();
```

Comment:
I think that's fine. As I see it, we should always reflect the active assignment. When we call `unsubscribe`, we leave the group, so the current assignment is no longer active. On the other hand, if we don't call `unsubscribe`, then the assignment will be active until the `onPartitionsRevoked` callback is triggered. Does that make sense or not?